src/pygeoprocessing/symbolic.py

```diff
@@ -188,10 +190,18 @@ def _generic_raster_op(*arg_list):
             raise ValueError(
                 "`target_nodata` is undefined (None) but there are nodata "
                 "values present in the input rasters.")
-        func_result = func(*[array[valid_mask] for array in array_list])
+        user_symbols = dict((symbol, array[valid_mask]) for (symbol, array) in
+                            zip(kwarg_names, array_list))
     else:
         # there's no nodata values to mask so operate directly
-        func_result = func(*array_list)
+        user_symbols = dict((symbol, array) for (symbol, array) in

```

Comment:
Since we're Python 3 now, can I request dictionary comprehension here?

```suggestion
        user_symbols = {(symbol, array) for (symbol, array) in
                        zip(kwarg_names, array_list)}
```

Reply:
Sure, happy to use a dict comprehension!  But the notation you mentioned will create a set of tuples and not a dict:

```
>>> type({(a, a) for a in range(5)})
<class 'set'>
>>> type({a: a for a in range(5)})
<class 'dict'>
```
I'll patch this shortly.

Reply:
Updated in 05b0195